### DISCUSSION TITLE
Feature request: Utilizing Empirical Mode Decomposition for pre-processing the data

### DISCUSSION THREAD

**Discussion by oshin94** [microsoft/farmvibes-ai#27] [Ideas]
Hi,
I have been going through the DeepMC part of the farmvibes.ai project and found that Wavelet Packet decomposition is used as a pre-processor to allow models to understand and fit the underlying patterns in the time-series data.
.
Similar to Wavelet Transform, Empirical Mode Decomposition (EMD) is an empirical, iterative and adaptive algorithm which decomposes a signal into components called Intrinsic Mode Functions (IMFs). EMD is also better suited for climate predictions on variables like temperature, rainfall etc., as it doesn't assume linearity of signal and has no prior assumptions of a basis function.
.
There are multiple papers that show EMD and its variants perform well along with LSTM as a hybrid model to forecast climate variables.
So, I was wondering if EMD would be a good additional option as a pre-processor along with the Wavelet Decomposition. Do let me know your thoughts. If you do feel it would be useful for the project, I will be happy to contribute the code to add that functionality.
.
References:

* Huang, N. E., Shen, Z., Long, S. R., Wu, M. C., Shih, H. H., Zheng, Q., Yen, N. C., Tung, C. C., & Liu, H. H. (1998). The empirical mode decomposition and the Hilbert spectrum for nonlinear and non-stationary time series analysis. Proceedings of the Royal Society of London. Series A: Mathematical, Physical and Engineering Sciences, 454(1971), 903–995. https://doi.org/10.1098/rspa.1998.0193

Thank you.

**Comment by peeyush-kumar** [comment]
EMD will be a good feature to have. Wavelets are used for scale based decomposition. One of the reasons is that it aids in explainability. As you correctly pointed out there are non-stationary aspects of the signals and boundary issues that still create challenges. It will be interesting to see how utilizing EMD solves for that.
